modules/custom/openy_daxko2/src/OpenyActivityFinderDaxkoBackend.php

```diff
@@ -0,0 +1,659 @@
+
+class OpenyActivityFinderDaxkoBackend extends OpenyActivityFinderBackend {
+
+  /**
+   * Daxko configuration.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $daxkoConfig;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * OpenyActivityFinderDaxkoBackend constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManager $entity_type_manager
+   *   The EntityTypeManager.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param CacheBackendInterface $cache
+   *   Cache default.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManager $entity_type_manager, CacheBackendInterface $cache) {
+    parent::__construct($config_factory);
+    $this->daxkoConfig = $config_factory->get('openy_daxko2.settings');
+    $this->entityTypeManager = $entity_type_manager;
+    $this->cache = $cache;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function runProgramSearch($parameters, $log_id) {
+    $parameters += [
+      'locations' => '',
+      'keywords' => '',
+      'next' => '',
+      'days' => '',
+      'ages' => '',
+    ];
+
+    $access_token = $this->getDaxkoToken();
+
+    $get = [];
+    $locationsConfig = $this->getDaxkoLocations();
+    $locationArgument = $parameters['locations'];
+    $locations = explode(',', $locationArgument);
+    $locations = array_unique($locations);
+    $locations = array_filter($locations);
+    if (empty($locationArgument)) {
+      $locations = array_keys($locationsConfig);
+    } elseif (count($locations) == count($locationsConfig)) {
+      // If all locations were selected we do not need to use
+      // location_ids argument.
+      $locationArgument = [];
+    }
+
+    $locations = array_filter($locations);
+
+    if (!empty($locations) && !empty($locationArgument)) {
+      $get['location_ids'] = implode(',', $locations);
+    }
+
+    $keywords = $parameters['keywords'];
+    if (!empty($keywords)) {
+      $get['keywords'] = $keywords;
+    }
+
+    if ($next = $parameters['next']) {
+      $get['after'] = $next;
+    }
+
+    if ($days = $parameters['days']) {
+      $get['days_offered'] = $days;
+    }
+
+    $ages = $parameters['ages'];
+    if (!empty($ages)) {
+      $ageGet = [];
+      foreach (explode(',', $ages) as $age) {
+        $ageGet[] = date('Y-m-d', strtotime('-' . $age . ' months'));
+      }
+      $get['birth_dates'] = implode(',', $ageGet);
+    }
+
+    if (isset($parameters['categories']) && !empty($parameters['categories'])) {
+      $get['category_ids'] = $parameters['categories'];
+    }
+
+    // Include facets. We need locations for Activity Finder.
+    $get['include_facets'] = TRUE;
+
+    $time_start = microtime(true);
+    $client = new Client(['base_uri' => $this->daxkoConfig->get('base_uri')]);
```

Comment:
we have http_client service in core - memory wise let's use it and avoid of creation new client for every request